modules/nlu-testing/src/views/full/confusionMatrix.tsx

```diff
@@ -0,0 +1,78 @@
+import _ from 'lodash'
+import React, { FC, Fragment, useEffect, useState } from 'react'
+
+import Plots from './plots'
+
+const ConfusionMatrix: FC<any> = props => {
+  const [CF, setCF] = useState({ intent: [], context: [], slot: [], slotCount: [] })
+
+  const createCM = (matrixType: 'slot' | 'intent' | 'context' | 'slotCount') => {
+    const datas: { predicted: string; expected: string }[] = props.dataResult[matrixType]
+    if (datas.length < 1) {
+      return []
+    }
+    const CM = {}
+    const allLabels = Array.from(new Set(datas.map(o => o.predicted).concat(datas.map(o => o.expected))))
+    for (const label of allLabels) {
+      CM[label] = {}
+      for (const subLabel of allLabels) {
+        CM[label][subLabel] = 0
+      }
+    }
+
+    for (const entry of datas) {
+      CM[entry.predicted][entry.expected] += 1
+    }
+
+    const z = []
```

Comment:
Please avoid single char variable. What does this refer to ? Usually you use single chars in for loops, map, forEach, etc. because it's pretty clear what that variable is